torch_xla/csrc/data_ops.cpp

```diff
@@ -291,4 +292,31 @@ std::vector<xla::XlaOp> BuildSplit(
   return splits;
 }
 
+xla::XlaOp BuildUpdateSlice(
+    const xla::XlaOp& input, const xla::XlaOp& source,
+    tensorflow::gtl::ArraySlice<const xla::int64> base_indices) {
+  xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::XlaOp reshaped_source =
+      XlaHelpers::NormalizeShape(source, input_shape.rank());
+  std::vector<xla::XlaOp> start_indices;
+  for (auto index : base_indices) {
+    start_indices.push_back(
+        XlaHelpers::ScalarValue<xla::int64>(index, input.builder()));
+  }
+  return xla::DynamicUpdateSlice(input, reshaped_source, start_indices);
+}
+
+xla::XlaOp BuildGenericSlice(
```

Comment:
I don't like this `Generic` cop-out name. Could this be `BuildReadSlice`? And maybe rename `BuildUpdateSlice` to `BuildWriteSlice` for better name duality.

Reply:
Renamed BuildGenericSlice to Build Slice.